pkg/controller/parent/controller.go

```diff
@@ -265,3 +270,66 @@ func (r *ParentReconciler) determineControllerType(ctx context.Context, u *unstr
 	}
 	return config.DefaultController, nil
 }
+
+// ReconcilePreviewMode reconciles the resource in preview mode
```

Comment:
Blocker: this doesn't feel maintainable, we now have two Reconcile methods.

Can we inject a fake CCC or use a similar existing mechanism?